test/e2e/apps/job.go

```diff
@@ -111,18 +110,20 @@ var _ = SIGDescribe("Job", func() {
 
 	// Pods sometimes fail, but eventually succeed, after pod restarts
 	ginkgo.It("should run a job to completion when tasks sometimes fail and are not locally restarted", func() {
+		// One failure, then a success, no local restarts.
+		// We can't use the random failure approach, because JobController
+		// will throttle frequently failing Pods of a given Job, ramping
+		// up to 6 minutes between restarts, making test timeout due to
```

Comment:
Nit: This comment says `JobController` will throttle up to 6 minutes between restarts, while the above comment says the Kubelet will throttle up to 5 minutes before restarts. Is one of these more accurate that the other?

Reply:
The above "local restarts" test set `restartPolicy` as `OnFailure` so it's kubelet that recreates the failed **container**, its max backoff period is indeed 5 minutes:
https://github.com/kubernetes/kubernetes/blob/429ad7db4f9e5a645d562548f514be94cd717c6c/pkg/kubelet/kubelet.go#L137

The "no local restarts" test set `restartPolicy` as `Never` so it's `JobController` that recreates the failed **Pod** of a given Job, its max backoff period is 6 minutes:
https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#pod-backoff-failure-policy
https://github.com/kubernetes/kubernetes/blob/429ad7db4f9e5a645d562548f514be94cd717c6c/pkg/controller/job/job_controller.go#L60

Reply:
Ah gotcha! This is great info - let's add it to the comment?

Reply:
Sorry, what info did you mean to add? The comment has included "kubelet will ... 5 minutes, JobController will ... 6 minutes", did you mean more details?